src/components/cylc/tree/cylc-tree.js

```diff
@@ -72,27 +72,113 @@ function computeCyclePointsStates (cyclePointNodes) {
 }
 
 /**
- * @param {FamilyProxyNode} left
- * @param {FamilyProxyNode} right
+ * The default comparator used to compare strings for cycle points, family proxies names,
+ * task proxies names, and jobs.
+ *
+ * @param left {string}
+ * @param right {string}
  * @returns {number}
+ * @constructor
+ */
+const DEFAULT_COMPARATOR = (left, right) => {
+  return left.toLowerCase()
+    .localeCompare(
+      right.toLowerCase(),
+      undefined,
+      {
+        numeric: true,
+        sensitivity: 'base'
+      }
+    )
+}
+
+/**
+ * Declare function used in sortedIndexBy as a comparator.
+ *
+ * @private
+ * @callback SortedIndexByComparator
+ * @param {object} leftObject - left parameter object
+ * @param {string} leftValue - left parameter value
+ * @param {object} rightObject - right parameter object
+ * @param {string} rightValue - right parameter value
+ * @returns {boolean} - true if leftValue is higher than rightValue
+ */
```

Comment:
Is this declaring an interface?

Reply:
Yes, kind of. A type that can be re-used in the docs. But won't affect the code, i.e. only for docs, linters, etc